### PR TITLE
Kh mobileapp question compatibility

### PR DESCRIPTION
**Pull request by kurvinh**
pushing work to stage-moodle35 for testing https://jira.2u.com/browse/CTED-368

### CODE REVIEW

File: javascript/mobile_questionnaire.js [hunk 1]
```diff
@@ -339,6 +387,171 @@ function textBoxObserver(mutationList, observer) {
...
+                    for (var x = 0; x < button.length; x++) {
+                        button[x].disabled = false;
+                    }
+                }
+                if (requiredInput === true && numberOfRequiredAnswers == finalRequiredInput && typeof(nextButton) != 'undefined') {
+                    for (var i = 0; i < nextButton.length; i++){
+                        nextButton[i].disabled = false;
+                    }
+                }
+            break;
```
Comment (JJSwanevelder) [inline javascript/mobile_questionnaire.js]: No sure if worth it adding default case statements. If you do not see a need for a default handle then this should be ok I guess.

File: javascript/mobile_questionnaire.js [hunk 2]
```diff
@@ -1,132 +1,180 @@
...
+	    var pageNum = pageNumArray[pageNumArray.length - 1].innerHTML;
+	    var button = document.getElementsByClassName('questionnaire submit-button');
+	    var disableSaveButtonFalse = document.getElementsByClassName('hidden-submit-button-check-false-' + pageNum);
+	    var allSliders = document.getElementsByClassName('range');
+	    var nextButton = document.getElementsByClassName('questionnaire next-button');
+	    var allCheckboxes = document.getElementsByClassName('item-checkbox');
+	    var allTextBoxes = document.getElementsByClassName('questionnaire-text');
+	    var backButton = document.getElementsByClassName('back-button');
+	    var dateTime = document.getElementsByClassName('datetime');
+        var select = document.getElementsByClassName('select');
```
[the remark this answers is not in the document]
Comment (kurvinh) [reply javascript/mobile_questionnaire.js]: shot

File: javascript/mobile_questionnaire.js [hunk 3]
```diff
@@ -1,132 +1,180 @@
...
+	        }
+	    }
+
+	    // Setting up observer for checkboxes.
+	    if (typeof(allCheckboxes) != 'undefined' && allCheckboxes.length > 0) { // NA onload.
+	        var observerOptions = {
+	            attributes: true,
+	            characterData: true,
+	        }
+	        for(var x = 0; x < allCheckboxes.length; x++) {
```
[the remark this answers is not in the document]
Comment (kurvinh) [reply javascript/mobile_questionnaire.js]: shot

File: lib.php [hunk 4]
```diff
@@ -994,7 +1031,28 @@ function save_questionnaire_data_branching($questionnaireid, $surveyid, $userid,
...
+                                        $dupecheck = $DB->get_record($responsetable,
+                                            ['response_id' => $rid,
+                                            'question_id' => $rquestionid]
+                                        );
+
+                                        if (empty($dupecheck)) {
+                                            $DB->insert_record($responsetable, $rec);
+                                        }
+                                    }
+                                }else {
```
[the remark this answers is not in the document]
Comment (kurvinh) [reply lib.php]: shot

File: javascript/mobile_questionnaire.js (hunk 1 above)
Comment (kurvinh) [reply javascript/mobile_questionnaire.js]: shot

Comment (kurvinh) [reply javascript/mobile_questionnaire.js]: I remember there being a stylistic thing to the codechecker, moodle does not want you to have a switch without a break statement

Comment (kurvinh) [reply javascript/mobile_questionnaire.js]: no need for a default statement, since I am trying to keep it to a single check for the observer

**Pull request merged**